walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private void addTimeRange(List<TimeRange> timeList, int startTime, int endTime, boolean includeEnd) {
+        timeList.add((TimeRange.fromStartEnd(  
+            startTime,
+            endTime,
+            includeEnd)));
+    }
+
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> eventTimeRanges = new ArrayList<>();
+        List<TimeRange> openTimeRanges = new ArrayList<>();
+
+        if(request.getAttendees() != null && !request.getAttendees().isEmpty()) {
+            if(TimeRange.WHOLE_DAY.duration() >= request.getDuration()) {
+                for(Event loopEvents: events) {
+                    eventTimeRanges.add((loopEvents.getWhen()));
+                }
+
+                Collections.sort(eventTimeRanges, TimeRange.ORDER_BY_START);
+
+                for(int i = 0; i < eventTimeRanges.size(); i++) {
+                    if (i == 0) {
+                        addTimeRange(
+                            openTimeRanges,
+                            TimeRange.START_OF_DAY,
+                            eventTimeRanges.get(i).start(),
+                            false
```

Comment:
If you need to use a literal as an argument, include the parameter name as an inline comment - in this case, that would look like `/* includeEnd= */ false`. This makes it much clearer when reading the code without the help of an IDE to populate the name (as during a pull request review). 

However, here you can probably do better. It would be even clearer to have two versions of the function which call a common parent, named something like `addTimeRangeWitgInclusiveEnd` and `addTimeRangeWithExclusiveEnd` (or your preferred names), but basically if you're just passing a boolean as a flag it is often more readable to have two versions of the function like that, which then call `addTimeRange` with the boolean set appropriately.